sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
+      device.setExternalFreeStorage(getUnusedExternalStorage(externalStorageStat));
+    }
+
+    DisplayMetrics displayMetrics = getDisplayMetrics();
+    if (displayMetrics != null) {
+      setScreenResolution(device, displayMetrics);
+      device.setWidthPixels(displayMetrics.widthPixels);
+      device.setHeightPixels(displayMetrics.heightPixels);
+      device.setScreenDensity(displayMetrics.density);
+      device.setScreenDpi(displayMetrics.densityDpi);
+    }
+
+    device.setBootTime(getBootTime());
+    device.setTimezone(getTimeZone());
+
+    return device;
+  }
+
+  @SuppressWarnings("deprecation")
+  private void setScreenResolution(Device device, DisplayMetrics displayMetrics) {
+    device.setScreenResolution(getResolution(displayMetrics));
+  }
+
+  private TimeZone getTimeZone() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+      LocaleList locales = context.getResources().getConfiguration().getLocales();
+      if (!locales.isEmpty()) {
+        Locale locale = locales.get(0);
+        return Calendar.getInstance(locale).getTimeZone();
+      }
+    }
+    return Calendar.getInstance().getTimeZone();
+  }
+
+  private Date getBootTime() {
+    // if user changes time, will give a wrong answer, consider ACTION_TIME_CHANGED
+    return new Date(System.currentTimeMillis() - SystemClock.elapsedRealtime());
+  }
+
+  private String getResolution(DisplayMetrics displayMetrics) {
+    int largestSide = Math.max(displayMetrics.widthPixels, displayMetrics.heightPixels);
+    int smallestSide = Math.min(displayMetrics.widthPixels, displayMetrics.heightPixels);
+    return largestSide + "x" + smallestSide;
+  }
+
+  /**
+   * Get MemoryInfo object representing the memory state of the application.
+   *
+   * @return MemoryInfo object representing the memory state of the application
+   */
+  private ActivityManager.MemoryInfo getMemInfo() {
+    try {
+      ActivityManager actManager = (ActivityManager) context.getSystemService(ACTIVITY_SERVICE);
+      ActivityManager.MemoryInfo memInfo = new ActivityManager.MemoryInfo();
+      if (actManager != null) {
+        actManager.getMemoryInfo(memInfo);
+        return memInfo;
+      }
+      log(SentryLevel.INFO, "Error getting MemoryInfo.");
+      return null;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting MemoryInfo.", e);
+      return null;
+    }
+  }
+
+  private Intent getBatteryIntent() {
+    return context.registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
+  }
+
+  /**
+   * Fake the device family by using the first word in the Build.MODEL. Works well in most cases...
+   * "Nexus 6P" -> "Nexus", "Galaxy S7" -> "Galaxy".
+   *
+   * @return family name of the device, as best we can tell
+   */
+  private String getFamily() {
+    try {
+      return Build.MODEL
+          .split(" ")[
+          0]; // might be possible to do it better, should we change that or keep compatibility?
```

Comment:
so the aim here is to find the "Family", which is Nexus/Galaxy and so on, as they have many versions.
eg "Galaxy S7" with the current approach becomes "Galaxy" which is right, but there are cases like this:
eg "Samsung Galaxy S7", so it becomes "Samsung" which is wrong, it should be "Galaxy" again.
some workarounds are to look for the manufacturer name, brand name, model name and ignore them and get the next String, so "Samsung Galaxy S7" will become "Galaxy" because we ignore the 1st String as it's the Build.MANUFACTURER and won't be the family name anyway, got it?

Reply:
will keep compatibility, for now, it will require more tests with different brands/models